src/metpy/plots/declarative.py

```diff
@@ -647,6 +651,35 @@ class MapPanel(Panel):
     This trait sets a user-defined title that will plot at the top center of the figure.
     """
 
+    @validate('area')
+    def _valid_area(self, proposal):
+        """Check that proposed string or tuple is valid and turn string into a tuple extent."""
+        area = proposal['value']
+
+        # Parse string, check that string is valid, and determine extent based on string
+        if isinstance(area, str):
+            region, modifier = self._parse_area_string(area)
```

Comment:
I think we could replace `_parse_area_string()` using a [regex](https://docs.python.org/3/library/re.html) with the `re` module:
```python
import re  # at the top
match = re.match(r'(\w+)([-+]*)$', area)
if match is None:
    # Throw error about bad area
else:
    region, modifier = match.groups()
```